### DISCUSSION TITLE
Customize the Apostrophe Login

### DISCUSSION THREAD

**Discussion by sakthivelsubbarayan96** [apostrophecms/apostrophe#2840] [Q&A] [answered]
How can I use my own API for apostrophe login?. I have a large number of users who can able to add/edit the apostrophe contents. The apostrophe has the own page the create the  Apostrophe users, But I have a large number of users existing in my own database, So if the user uses the apostrophe login page after the user enter the details I have an own API to authenticate the user and assign the permission for add/edit the contents. Is this possible?

**Comment by abea** [comment, marked as answer]
The are a couple of single sign-on modules for Apostrophe that could help: https://apostrophecms.com/extensions?extension-category=single-sign-on

[The Passport module](https://www.npmjs.com/package/apostrophe-passport) is probably your best bet. You  may have to write your own Passport "strategy," or you might be able to use an existing one if you're already using SSO.
